src/tests/test_gsp.py

```diff
@@ -418,3 +420,185 @@ def test_read_truths_for_all_gsp_compact(db_session, api_client):
     assert len(datetimes_with_gsp_yields[0].generation_kw_by_gsp_id) == 1
     assert len(datetimes_with_gsp_yields[1].generation_kw_by_gsp_id) == 2
     assert len(datetimes_with_gsp_yields[2].generation_kw_by_gsp_id) == 1
+
+
+def test_is_fake_specific_gsp(monkeypatch, db_session, api_client, gsp_id=1, pytest_print=None):
+    """### Test FAKE environment specific _gsp_id_ routes are populating
+    with fake data.
+
+    #### Parameters
+    - **gsp_id**: Please set to any non-zero integer that is <= GSP_TOTAL
+    - **pytest_print**: If you'd like to inspect the forecast values
+    in the FAKE environment, please use 'pytest -s' in the CLI to inspect
+    the values. Using the -s switch should enable the print statement.
+    Otherwise can use 'pytest -v' and it will run the tests as normal and
+    suppress the verbose print statements.
+    """
+
+    monkeypatch.setenv("FAKE", "1")
+    assert is_fake() == 1
+    # Specific _gsp_id_ route/endpoint for successful connection
+    response = api_client.get(f"/v0/solar/GB/gsp/{gsp_id}/forecast")
+    assert response.status_code == 200
+
+    forecasts = make_fake_forecast(
+        gsp_id=gsp_id,
+        session=db_session,
+        forecast_values=None,
+        forecast_values_latest=None,
+        add_latest=True,
+        historic=True,
+        model_name="fake_model",
+    )
+    db_session.add(forecasts)
+    db_session.commit()
```

Comment:
remove adding forecasts in tests, this  is already done in the code you put into gsp.py

Reply:
I've removed the calls to the DB in the test cases. Could you review, and let me know if I've done it correctly, please?

Reply:
Have you? I can still see the